### PR TITLE
Make autorest.go version configurable.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Added command flag 'a' to optionally specify autoret.go version.

Thanks you for your contribution to the Azure-SDK-for-Go! We will triage and review it as quickly as we can.

As part of your submission, please make sure that you can make the following assertions:

 - [ ] I'm not making changes to Auto-Generated files which will just get erased next time there's a release.
   - If that's what you want to do, consider making a contribution here: https://github.com/Azure/autorest.go
 - [ ] I've tested my changes, adding unit tests where applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, or I'm fixing a bug that warrants its own release and I'm targeting the `master` branch.
 - [ ] If I'm targeting the `master` branch, I've also added a note to [CHANGELOG.md](https://github.com/Azure/azure-sdk-for-go/blob/master/README.md).
 - [ ] I've mentioned any relevant open issues in this PR, making clear the context for the contribution.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Ping @marstr

File: tools/generator/program.go [hunk 1]
```diff
@@ -42,11 +42,16 @@ import (
 	"github.com/marstr/collection"
 )
 
+const (
+	defaultGenVer = "@microsoft.azure/autorest.go@v2.0.41"
```
Comment (marstr) [inline tools/generator/program.go]: I'd say if it's unspecified, we shouldn't provide the "use" flag at all. That way we use whatever the most recent release of autorest.go has been published.

Comment (jhendrixMSFT) [reply tools/generator/program.go]: Do we emit the version used anywhere so there's a record?  I'd hate to leave it up to fate so we can't recreate how an SDK was generated.

Comment (marstr) [reply tools/generator/program.go]: @jhendrixMSFT and I chatted offline, we're going to pull out my suggestion for now, and fix it when submit a seperate PR to brand the version of Autorest.Go that was used. Once that happens, we'll take a look at what the default behavior here should be.

**Pull request merged**